integrations/gitlab_v2/client.py

```diff
@@ -0,0 +1,296 @@
+    @cache_iterator_result()
+    async def get_groups(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for groups in self._make_paginated_request(self.groups_url):
+            yield groups
+
+    async def get_merge_requests(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for merge_requests in self._make_paginated_request(
+            self.merge_requests_url
+        ):
+            merge_requests_with_projects = await asyncio.gather(
+                *[
+                    self._enrich_merge_request_with_project(merge_request)
+                    for merge_request in merge_requests
+                ]
+            )
+
+            yield merge_requests_with_projects
+
+    async def get_merge_request(
+        self, project_id: int, merge_request_id: int
+    ) -> dict[str, Any]:
+        return await self._make_request(
+            url=f"{self.projects_url}/{project_id}/merge_requests/{merge_request_id}"
+        )
+
+    async def get_issues(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for issues in self._make_paginated_request(self.issues_url):
+            issues_with_projects = await asyncio.gather(
+                *[self._enrich_issues_with_project(issue) for issue in issues]
+            )
+
+            yield issues_with_projects
+
+    async def _create_project_hook(self, app_host: str) -> None:
+        gitlab_project_webhook_host = f"{app_host}/integration/webhook"
+        async for projects in self.get_projects():
+            # Create webhooks concurrently for each project
+            await asyncio.gather(
+                *[
+                    self._process_project_hooks(project, gitlab_project_webhook_host)
+                    for project in projects
+                ]
+            )
+
+    async def _process_project_hooks(
+        self, project: dict[str, Any], webhook_host: str
+    ) -> None:
+        try:
+            hooks = await self._get_project_hooks(project["id"])
+
+            # Create or skip the project hook
+            await self._create_or_skip_project_hook(project, hooks, webhook_host)
+
+        except Exception as e:
+            logger.error(
+                f"Error processing hooks for project {project['path_with_namespace']}: {e}"
+            )
+
+    async def _create_or_skip_project_hook(
+        self, project: dict[str, Any], hooks: list[dict[str, Any]], webhook_host: str
+    ) -> None:
+        if any(hook["url"] == webhook_host for hook in hooks):
+            logger.info(
+                f"Skipping hook creation for project {project['path_with_namespace']}"
+            )
+            return
+
+        payload: dict[str, Any] = {
+            "id": project["id"],
+            "name": f"{ocean.config.integration.identifier}-{WEBHOOK_NAME}",
+            "url": webhook_host,
+            **WEBHOOK_EVENTS_TO_TRACK,
+        }
+
+        try:
+            logger.info(f"Creating hook for project {project['path_with_namespace']}")
```

Comment:
This logic will be best separated clearly. Leave only logics that deal with get and create here, move the logic that compares and looks for the existence of the webhook into main